image_provisioner/playbooks/roles/docker-config/tasks/main.yaml

```diff
@@ -12,16 +21,61 @@
 - name: install docker
   yum: name=docker state=latest
 
-- name: check for existing PV
-  command: pvdisplay
-  register: pv_result
+- block:
+   # should be used until the mirrors are fixed to install right container-selinux package
+   - name: remove existing container-selinux package
+     yum: name=container-selinux state=absent
+
+   - name: install container-selinux-1.12.5-14
+     yum: name=container-selinux-1.12.5-14.el7 state=latest
+
+   - name: create a xfs filesystem
+     shell: mkfs.xfs -f /dev/xvdb
+
+   - name: update fstab
+     shell: echo "/dev/xvdb /var/lib/docker                    xfs   defaults        0 0" >> /etc/fstab
+
+  when: docker_storage_driver == "overlay"
+
+- block:
+   - name: restart machine
```

Comment:
Oh I see it's an indentation and spacing issue with the when at the end of the block. Thanks @mffiedler